### ISSUE TITLE
[bug] Unable to subscribe to push notifications with Toot! on iOS

### ISSUE THREAD

**Issue by QuietMisdreavus** [superseriousbusiness/gotosocial#1139]
### Describe the bug with a clear and concise description of what the bug is.

Mastodon-client apps on iOS do not seem to deliver push notifications for a GTS instance i recently joined. Metatext popped some error dialogs upon first signing in (that i cannot reproduce), but Toot! has a specific button to enable push notifications, which returns a 404:

![iOS dialog saying "Failed to enable notifications: Server did not return an OK response code (404)"](https://user-images.githubusercontent.com/5217170/203837946-8761b764-61e9-4113-8ed7-bd125fa90621.jpeg)

### What's your GoToSocial Version?

GoToSocial 0.5.2 git-c31f219

### GoToSocial Arch

(unknown - i am not the admin of this instance)

### What happened?

The "Notifications" feed loads fine while in the app (both Metatext and Toot!), but iOS apps do not generate push notifications this way, so it requires manually checking the feed to see notifications of any kind.

### What you expected to happen?

Expected the app to subscribe to push notifications similarly to how Mastodon instances can.

### How to reproduce it?

1. Use Toot! to sign in to a GTS instance.
2. Open the in-app settings menu, either through the three-dot menu in the corner, or via the "Settings" link in your profile view.
3. Scroll to the "Notifications" heading.
4. Tap "Enable Notifications".

**Comment by tsmethurst** [comment]
We don't implement push notifications yet so that makes sense :D

**Comment by QuietMisdreavus** [comment]
That makes sense! Thanks for the update ❤️

**Comment by whtiehack** [comment]
hi, is there any update on this now?

**Comment by tsmethurst** [comment]
> hi, is there any update on this now?

nope! we still don't implement them, and it's not on the roadmap; likely this is a nice-to-have that we'll work on once we're in beta

**Comment by daenney** [comment]
We don't have push notification support, so it's not so much a bug. I'm going to close this one in favour of #1350, as that's the feature request for WebPush which would resolve this. Feel free to subscribe to that one if you'd like to get updates when work starts on that. Though it's probably going to be a while longer still :sweat_smile:.
